### PR TITLE
Initial commit of SOFT/HRUF Splitography keyboard

### PR DESCRIPTION
**Pull request by analyticd**
Add SOFT/HRUF Splitography keyboard and several keymaps for it.

## Description

Documentation for all but one of the keymaps, default_with_ctl_shft_alt_switched, can be found at 

http://thedarnedestthing.com/search?query=qmk

This is not my site or keymaps with the exception of default_with_ctl_shft_alt_switched.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Checklist

- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [ ] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Comment by yanfali** [comment]
Can you move a lot of this common code up to the keyboard level? having it all in live in the keymaps means you have a lot of duplicate code and that makes it inaccessible to configurator. Basically this keyboard will be unusable from a configurator build.

File: keyboards/splitography/keymaps/beaklmu/common/init.h [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+// ....................................................................... Audio
+
+void matrix_init_user(void)
+{
+#ifdef STENO_ENABLE
+  steno_set_mode(STENO_MODE_BOLT);          // or STENO_MODE_GEMINI
+#endif
+#ifdef AUDIO_ENABLE
+  startup_user();
+#endif
```
Comment (drashna) [inline keyboards/splitography/keymaps/beaklmu/common/init.h]: This is handled by `quantum.c` (core), and isn't needed.  In fact, this overrides default behavior.

File: keyboards/splitography/keymaps/beaklmu/common/keycode_functions.h [hunk 2]
```diff
@@ -0,0 +1,660 @@
+
+// .......................................................... Keycode Primitives
+
+// register simple key press
+void tap_key(uint16_t keycode)
+{
+  register_code  (keycode);
+  unregister_code(keycode);
+}
```
Comment (drashna) [inline keyboards/splitography/keymaps/beaklmu/common/keycode_functions.h]: This is not needed.  `tap_code` is a core feature/function that does the same thing.

File: keyboards/splitography/keymaps/beaklmu/common/keycode_functions.h [hunk 3]
```diff
@@ -0,0 +1,660 @@
...
+
+void shift_key(uint16_t keycode)
+{
+  register_code  (KC_LSFT);
+  tap_key        (keycode);
+  unregister_code(KC_LSFT);
+}
```
Comment (drashna) [inline keyboards/splitography/keymaps/beaklmu/common/keycode_functions.h]: Also, not needed.  `tap_code16` handles modded characters.

Eg, `tap_code16(S(keycode))` would do the same thing.

File: keyboards/splitography/keymaps/beaklmu/common/keycode_functions.h [hunk 4]
```diff
@@ -0,0 +1,660 @@
...
+        shift_key(keycode);
+      }
+      else {
+        tap_key(keycode);
+      }
+      return true;
+    }
+  }
+  return false;
+}
```
Comment (drashna) [inline keyboards/splitography/keymaps/beaklmu/common/keycode_functions.h]: This looks like it's replicating the "Auto Shift" function? 

https://docs.qmk.fm/#/feature_auto_shift

File: keyboards/splitography/keymaps/beaklmu/common/keycode_functions.h [hunk 5]
```diff
@@ -0,0 +1,660 @@
...
+
+// ..................................................................... Keymaps
+
+// void persistant_default_layer_set(uint16_t default_layer)
+// {
+//   eeconfig_update_default_layer(default_layer);
+//   default_layer_set            (default_layer);
+// }
+
+void clear_layers(void)
```
Comment (drashna) [inline keyboards/splitography/keymaps/beaklmu/common/keycode_functions.h]: `layer_move(0)` does the same thing.

Comment (analyticd) [reply keyboards/splitography/keymaps/beaklmu/common/keycode_functions.h]: Thank you for your very helpful responses. I hope to have time to refactor this and resubmit in a different pull request. I will close this now, but I thank you for your detailed response. I appreciate your time and help.

**Pull request closed without merging**